keel-echo/src/main/kotlin/com/netflix/spinnaker/keel/echo/UnhealthyNotificationListener.kt

```diff
@@ -1,37 +1,55 @@
 package com.netflix.spinnaker.keel.echo
 
+import com.netflix.spinnaker.config.UnhealthyNotificationConfig
 import com.netflix.spinnaker.keel.api.AccountAwareLocations
 import com.netflix.spinnaker.keel.api.Locatable
 import com.netflix.spinnaker.keel.api.Monikered
 import com.netflix.spinnaker.keel.api.Resource
+import com.netflix.spinnaker.keel.events.ClearNotificationEvent
 import com.netflix.spinnaker.keel.events.ResourceHealthEvent
 import com.netflix.spinnaker.keel.events.NotificationEvent
 import com.netflix.spinnaker.keel.notifications.ClusterViewParams
 import com.netflix.spinnaker.keel.notifications.Notification
 import com.netflix.spinnaker.keel.notifications.NotificationScope.RESOURCE
 import com.netflix.spinnaker.keel.notifications.NotificationType.UNHEALTHY_RESOURCE
+import com.netflix.spinnaker.keel.notifications.friendlyDuration
+import com.netflix.spinnaker.keel.persistence.UnhealthyRepository
 import com.netflix.spinnaker.keel.veto.unhealthy.UnsupportedResourceTypeException
 import org.springframework.beans.factory.annotation.Value
+import org.springframework.boot.context.properties.EnableConfigurationProperties
 import org.springframework.context.ApplicationEventPublisher
+import org.springframework.context.annotation.Configuration
 import org.springframework.context.event.EventListener
 import org.springframework.core.env.Environment
 import org.springframework.stereotype.Component
+import java.time.Duration
 
-
+@Configuration
+@EnableConfigurationProperties(UnhealthyNotificationConfig::class)
 @Component
 class UnhealthyNotificationListener(
+  private val config: UnhealthyNotificationConfig,
+  private val unhealthyRepository: UnhealthyRepository,
```

Comment:
we now have a repo to store the time we notice a resource became unhealthy

Reply:
Sometimes I worry a little about proliferation of repositories and the complexity that brings, but I'll go along for the ride here and just keep an eye out for complexity growing...

Reply:
I personally don't think having lots of simple SQL tables is complex. The complexity I worry about it a table doing two different things, since that is really hard to refactor or use or optimize